cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -509,3 +511,96 @@ func (s *Server) CreateInstalledPackage(ctx context.Context, request *corev1.Cre
 		InstalledPackageRef: installedRef,
 	}, nil
 }
+
+// UpdateInstalledPackage Updates an installed package managed by the 'kapp_controller' plugin
+func (s *Server) UpdateInstalledPackage(ctx context.Context, request *corev1.UpdateInstalledPackageRequest) (*corev1.UpdateInstalledPackageResponse, error) {
+	log.Infof("+kapp-controller UpdateInstalledPackage")
+
+	// Validate the request
+	if request == nil || request.GetInstalledPackageRef() == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "no request AvailablePackageRef provided")
+	}
+	if request.GetInstalledPackageRef().GetContext().GetNamespace() == "" || request.GetInstalledPackageRef().GetIdentifier() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "required context or identifier not provided")
+	}
+	if request.GetInstalledPackageRef().GetIdentifier() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "no request Name provided")
+	}
+
+	packageRef := request.GetInstalledPackageRef()
+	cluster := packageRef.GetContext().GetCluster()
+	reconciliationOptions := request.GetReconciliationOptions()
+	pkgVersion := request.GetPkgVersionReference().GetVersion()
+	installedPackageName := request.GetInstalledPackageRef().GetIdentifier()
+	values := request.GetValues()
+	targetNamespace := request.GetInstalledPackageRef().GetContext().GetNamespace()
+
+	if cluster == "" {
+		cluster = s.globalPackagingCluster
+	}
+
+	typedClient, _, err := s.GetClients(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	// fetch the package install
+	pkgInstall, err := s.getPkgInstall(ctx, cluster, targetNamespace, installedPackageName)
+	if err != nil {
+		return nil, errorByStatus("get", "PackageInstall", installedPackageName, err)
+	}
+
+	// Update the values.yaml values file if any is passed, otherwise, delete the values
+	if values != "" {
+		secret, err := s.newSecret(installedPackageName, values, targetNamespace)
+		if err != nil {
+			return nil, errorByStatus("upsate", "Secret", secret.Name, err)
+		}
+		updatedSecret, err := typedClient.CoreV1().Secrets(targetNamespace).Update(ctx, secret, metav1.UpdateOptions{})
+		if updatedSecret == nil || err != nil {
+			return nil, errorByStatus("update", "Secret", secret.Name, err)
+		}
+	} else {
+		// Delete all the associated secrets
```

Comment:
Hmm. Unsure about this but don't see a better option. We don't want to orphan secrets created by us, but neither would I want to delete secrets that are not created by us. Perhaps eventually we can label secrets created by Kubeapps and only delete them if it has the label.